src/moleculesampleapp/BoxSurface.java

```diff
@@ -0,0 +1,63 @@
+package moleculesampleapp;
+
+import com.sun.javafx.geom.Vec3d;
+import javafx.scene.Group;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.PhongMaterial;
+import javafx.scene.shape.Box;
+import javafx.scene.transform.Translate;
+
+import java.util.ArrayList;
+
+/**
+ * Created by Piotr on 06/02/2016.
+ */
+public class BoxSurface {
+    protected ArrayList<VectorPoint> points;
+    public VectorPoint min, max;
+    protected Vec3d pitch;
+    protected double pitchScale;
+
+    public BoxSurface() {
+        this(new Vec3d(10, 10, 10), 1.0);
+    }
+
+    public BoxSurface(Vec3d pitch_, double pitchScale_) {
+        points = new ArrayList<>();
+        min = new VectorPoint();
+        max = new VectorPoint();
+        pitch = pitch_;
+        pitchScale = pitchScale_;
+    }
+
+    public void add(VectorPoint point) {
+        if (points.isEmpty()) {
+            max = new VectorPoint(point);
+            min = new VectorPoint(point);
+        } else {
+            if (max.position.x < point.position.x) max.position.x = point.position.x;
+            if (max.position.y < point.position.y) max.position.y = point.position.y;
+            if (max.position.z < point.position.z) max.position.z = point.position.z;
+
+            if (max.vector.x < point.vector.x) max.vector.x = point.vector.x;
```

Comment:
there is a collection of VectorPoints (protected ArrayList<VectorPoint> points;) -> better  to use map() and Collections.max()